### PR TITLE
Update http.py

### PR DESCRIPTION
**Pull request by asterized**
Fixed problem in HTTPSession, renamed HTTPSession to _Session

### CODE REVIEW

**Comment by SomeHybrid** [comment]
list of fixes:
1: changed `self.close()` to `loop.run_until_complete(self.close())` as `aiohttp.ClientSession.close()` is an async function
2: initialized `aiohttp.ClientSession` in an async function to prevent DeprecationWarning as you need to initialize it in an async function

**Comment by SomeHybrid** [comment]
cool formatting

File: bot/util/http.py [hunk 1]
```diff
@@ -61,22 +62,20 @@ async def inner(*args, no_cache=False, **kwargs):
...
-            self.close()
+            _loop.run_until_complete(self.close())
+
+
+async def generate_session():
+    return _Session()
 
 
-session = HTTPSession()
+session = _loop.run_until_complete(generate_session())
```
Comment (sexnine) [inline bot/util/http.py:78]: This does not need to be async as initializing an object is sync.

```suggestion
session = _Session()
```

**Review by sexnine — CHANGES_REQUESTED** [review]
@nuke886 since your the one utilizing this class.  I'll leave you to deal with this, feel free to merge once your happy :)

File: bot/util/http.py (hunk 1 above)
Comment (SomeHybrid) [reply bot/util/http.py:78]: initializing a ClientSession in a non-async function returns a DeprecationWarning

**Comment by SomeHybrid** [comment]
Details for more formatting as forgor to put it in the description

Added a Session class that returns _Session and i've made a mistake again sad

File: bot/util/http.py (hunk 1 above)
Comment (nuke886) [reply bot/util/http.py]: `asyncio.get_event_loop()` was Deprecated in Python 3.10

**Pull request closed without merging**